.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -euo pipefail
+
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "temp-fleet-server" && "$BUILDKITE_STEP_KEY" == "publish" ]]; then #should be changed to "fleet-server" after tests
+  export DOCKER_USER=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
+  export DOCKER_USER_SECRET=$(retry 5 vault kv get -field password "${DOCKER_REGISTRY_SECRET_PATH}")
```

Comment:
Thinking out loud, what do we think if this does not export the secret at all, but rather performs directly the docker_login?

Reply:
Indeed that would be similar since that hook runs first for each step :+1: 
If another step needs this docker_login (e2e tests ? don't know) we can create a function here and reuse it in all the "ifs" needed.

The only disadvantage of doing the docker login here  I think it is we're coupling on how Buildkite works. But I would say it is ok, we shouldn't move to another CI framework anytime soon :D , and we take advantage of Buikdkite

Reply:
> we shouldn't move to another CI framework anytime soon 

I hope not!!!!!

Reply:
That's a good idea, but unfortunately, it doesn't work. I have the "access denied" error. It looks like pre-command is working somewhere in the dedicated environment but we need to push images from the particular agent and we have to run docker_login in this particular agent to establish a connection with the docker registry.

Reply:
will try it once again, I found one mistake...

Reply:
the result of tests:
`docker login` in the pre-command os OK
`docker logout` in the pre-exit -  `docker: command not found`
@alexsapran @mrodm @leo-ri

Reply:
I tied to use post-command as well, but unfortunately, the same error:
`.buildkite/hooks/post-command: line 6: docker: command not found`